### PR TITLE
added history endpoints

### PR DESCRIPTION
**Pull request by Huzzy619**
Though some dependencies like "fastapi_users" need to be installed.

### CODE REVIEW

**Comment by contactodohgerald** [comment]
@Huzzy619 nice job man,

 i go come for private lessons if you will have me

File: app/db/models/history.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from typing import Optional
+from sqlmodel import Field, Relationship, SQLModel
+from .common import TimestampModel
+from .user import User
+
+class History(SQLModel, table= True):
+    __tablename__ = "history"
+    
+    id: Optional[int] = Field(default=None, primary_key=True)
```
Comment (a11rew) [inline app/db/models/history.py]: For consistency, please update this to inherit the common uuid model like the rest of the models.

File: app/db/models/history.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    input: str
+    result: str
+    user_id: Optional[int] = Field(default=None, foreign_key="user.id")
+
+    user: Optional[User] = Relationship(back_populates="user")
+    created_at: Optional[TimestampModel.created_at]
```
Comment (a11rew) [inline app/db/models/history.py]: We would need `created_at` to not be optional, `updated_at` maybe not so much but for consistency with the other models just inherit everything from the `TimestampModel`

File: app/api/history/views.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+@router.get("/user/history", response_model=list[GetHistory])
+async def get_user_history(offset: int = 0, limit: int = Query(default=20, lte=20), session: AsyncSession = Depends(db_session)) -> list[History]:
+    user_history = HistoryService(session=session)
+    user = current_user
+    return await user_history.get_user_history(user_id= user.id, limit=limit, offset=offset)
+
+
+@router.delete("/remove/history", response_model=GetHistory)
+async def remove_all_history(session: AsyncSession = Depends(db_session)):
```
Comment (a11rew) [inline app/api/history/views.py]: I can't seem to think of a situation where we would want to allow deletion of the history records from the outside. Could you give some insight into your decision to support deletion?

File: app/db/models/history.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+from sqlmodel import Field, Relationship, SQLModel
+from .common import TimestampModel
+from .user import User
+
+class History(SQLModel, table= True):
+    __tablename__ = "history"
+    
+    id: Optional[int] = Field(default=None, primary_key=True)
+    input: str
+    result: str
```
Comment (a11rew) [inline app/db/models/history.py]: See the definition for the [search endpoint here](https://github.com/a11rew/vip-service-backend/blob/616256c9d6934f9235960b44391441aa3729eea1/app/api/search/views.py#L9)

We accept name, gender, occupation, and age and return the same keys but with an is_vip property included. 

All of those values would not fit in a single str field no?

File: app/api/history/views.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+current_user = fastapi_users.current_user()
+
+router = APIRouter(
+    dependencies= [Depends(current_user)]
+)
+
+
+@router.get("/user/history", response_model=list[GetHistory])
+async def get_user_history(offset: int = 0, limit: int = Query(default=20, lte=20), session: AsyncSession = Depends(db_session)) -> list[History]:
```
Comment (a11rew) [inline app/api/history/views.py:18]: Nice, pagination, looks good :+1:

**Comment by Huzzy619** [comment]
> @Huzzy619 nice job man,
> 
> i go come for private lessons if you will have me

Abeg o, me sef dey struggle with am.

**Pull request closed without merging**